### PR TITLE
Fix test warning

### PR DESCRIPTION
**Pull request by jasonjunweilyu**
This PR is to fix test warnings, either by changing them to report errors when warnings are not issued if they are designed in fab with intentions through pytest.warns() or by filtering them out if they are related to deprecation warnings in dependent packages with @pytest.mark.filterwarnings(). The motivation of this PR is explained in Issue https://github.com/hiker/fab/issues/5. The fix for test errors is in another PR. 

This PR has passed the CI pipeline. It was tested to have no warnings with Python 3.12. However, there are still some Python-version-specific warnings left if you look at the CI pipeline results. The related PR on hiker/fab is here: https://github.com/hiker/fab/pull/10

Depends on
* [x] #291.

### CODE REVIEW

File: tests/system_tests/psyclone/skeleton/algorithm/algorithm_mod.x90
Comment (MatthewHambley) [inline tests/system_tests/psyclone/skeleton/algorithm/algorithm_mod.x90:1]: I seem to recognise this change from a different PR.

File: tests/system_tests/psyclone/test_psyclone_system_test.py
Comment (MatthewHambley) [inline tests/system_tests/psyclone/test_psyclone_system_test.py:1]: This also looks familiar although it has new stuff too. Has there been a branching issue?

File: tests/unit_tests/steps/grab/test_svn_fcm_unit_test.py
Comment (MatthewHambley) [inline tests/unit_tests/steps/grab/test_svn_fcm_unit_test.py:1]: Another duplicate change.

File: tests/unit_tests/steps/test_psyclone_unit_test.py
Comment (MatthewHambley) [inline tests/unit_tests/steps/test_psyclone_unit_test.py:1]: Seen in a different PR.

**Review by MatthewHambley — CHANGES_REQUESTED** [review]
A lot of the log chatter is probably left over development nonsense however this change does neatly document what exists.

**Comment by hiker** [comment]
Hi Matthew,
thanks a lot for the very quick review. Note that Jason is on leave, and I've asked him to put this PR in before he goes, but this PR needs to be applied after #291 (since without this patch we can't get CI to give us all green on our system, so it is dependent on that). Unfortunately, we can't (currently) set a label to indicate which PR is ready and which isn't :( I will try to fix #291 for Jason, and update this PR then.

**Comment by jasonjunweilyu** [comment]
Hi @MatthewHambley , I have merged `metomi:master` into `hiker:fix_test_warning` after the pull request of merging `hiker:fix_test_error` into `metomi:master` was approved. Now the duplicated change issues should have all gone, and this pull request is ready for review. Thanks.

**Review by MatthewHambley — APPROVED** [review]
I think this is ready for trunk.

**Pull request merged**